Core/include/Acts/EventData/detail/GenerateParameters.hpp

```diff
@@ -165,14 +168,17 @@ inline double generateQoverP(generator_t& rng,
       options.charge,
       -options.charge,
   };
-  UniformReal pDist(options.pMin, options.pMax);
+  UniformReal pDist(
+      options.pLogUniform ? std::log(options.pMin) : options.pMin,
+      options.pLogUniform ? std::log(options.pMax) : options.pMax);
 
   // draw parameters
   const std::uint8_t type = particleTypeChoice(rng);
   const double q = qChoices[type];
 
-  const double p =
-      pDist(rng) * (options.pTransverse ? 1. / std::sin(theta) : 1.);
+  double p = pDist(rng);
+  p = options.pLogUniform ? std::exp(p) : p;
```

Comment:
Now, we have 2 places, where we (need to) inject information about the log-distribution.

Maybe, it could be more transparent if we make 2 separate branches?
```
double p = 0.;
if (options.pLogUniform) {
  ...
} else {
  ...
}

if (options.pTransverse) {
  p /= std::sin(theta);
}
```

Reply:
I don't know, looks similarly complicated to me